input/Glfw/GlfwInputDevice.cpp

```diff
@@ -21,17 +22,32 @@ namespace NovelRT::Input::Glfw
             throw Exceptions::InitialisationFailureException("GLFW3 failed to initialise.", std::string(output));
         }
 
-        if (window == nullptr)
+        if (device == nullptr)
         {
             throw Exceptions::NullPointerException(
-                "Could not initialise GLFW input service - null pointer was provided for window.");
+                "Could not initialise GLFW input service - null pointer was provided for windowing device.");
         }
 
         _logger.logInfoLine("Initialising GLFW input service.");
         _availableKeys = std::map<std::string, NovelKey>();
         _mappedActions = std::vector<InputAction>();
-        _window = reinterpret_cast<GLFWwindow*>(window);
-
+        _inputBufferCount = DEFAULT_INPUT_BUFFER_COUNT;
+        _keyStates = std::vector<std::unordered_map<int32_t, KeyStateFrameChangeLog>>(_inputBufferCount);
+        _previousBufferIndex = 0;
+        _currentBufferIndex = 1;
+
+        auto properDevice = reinterpret_cast<NovelRT::Windowing::Glfw::GlfwWindowingDevice*>(device.get());
```

Comment:
The cast should most likely be:
```suggestion
        auto properDevice = std::reinterpret_pointer_cast<NovelRT::Windowing::Glfw::GlfwWindowingDevice>(device);
```

As a side note, are we concerned about someone not passing in a GLFW windowing device here? do we want to sanitise this or is that too slow?

Reply:
I don't believe this is a concern, nor do we really want to sanitise here.